lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -379,6 +379,31 @@ class DecomposeAtenMatmulOp : public OpRewritePattern<AtenMatmulOp> {
 };
 } // namespace
 
+namespace {
+class DecomposeAtenTOp : public OpRewritePattern<AtenTOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenTOp op,
+                                PatternRewriter &rewriter) const override {
+    Value lhs = op.self();
+    int lhsRank = getTensorRank(lhs);
```

Comment:
`getTensorRank` returns an int because it uses -1 as an error value. I think it's fine to leave the int, but it would be good to add a check making sure it's non-negative